kedro/framework/project/__init__.py

```diff
@@ -126,12 +126,7 @@ def _load_data(self):
                 self._pipelines_module
             )
         except (ModuleNotFoundError, AttributeError) as exc:
-            # for backwards compatibility with templates < 0.17.2
-            # where no pipelines_registry is defined
-            if self._pipelines_module in str(exc):  # pragma: no cover
-                project_pipelines = {}
-            else:
-                raise
+            raise exc
```

Comment:
I think we can get rid of the whole `try/except/else` here now and just do this:
```
register_pipelines = self._get_pipelines_registry_callable(self._pipelines_module)
project_pipelines = register_pipelines()
```
But check with @lorenabalan since I might have forgotten what we were doing here 😅

Reply:
Yes that's right I was just about to comment that.